source/digits_hits/src/GateRestrictedVolume.cc

```diff
@@ -0,0 +1,67 @@
+/** ----------------------
+  Copyright (C): OpenGATE Collaboration
+  This software is distributed under the terms
+  of the GNU Lesser General  Public Licence (LGPL)
+  See LICENSE.md for further details
+  ----------------------*/
+
+#include "GateRestrictedVolume.hh"
+
+#include "G4UnitsTable.hh"
+
+#include "GateRestrictedVolumeMessenger.hh"
+#include "GateTools.hh"
+
+GateRestrictedVolume::GateRestrictedVolume(GatePulseProcessorChain* itsChain,
+                                       const G4String& itsName)
+  : GateVPulseProcessor(itsChain, itsName)
+{
+  m_messenger = new GateRestrictedVolumeMessenger(this);
```

Comment:
```suggestion
  m_messenger = std::make_unique<GateRestrictedVolumeMessenger>(this);
```
I guess if m_messenger is defined as unique_ptr then you need something like this.

Reply:
Unique pointer issue has been already solved